src/redprl/refiner_inductive_spec.fun

```diff
@@ -0,0 +1,360 @@
+functor RefinerIndSpec (Sig : MINI_SIGNATURE) =
+struct
+  structure Kit = RefinerKit (Sig)
+  structure ComRefinerKit = RefinerCompositionKit (Sig)
+  open RedPrlAbt Kit ComRefinerKit
+  type hyp = Sym.t
+  infixr @@
+  infix 1 || #>
+  infix 2 >> >: >:? >:+ $$ $# // \ @>
+
+  structure InductiveSpec =
+  struct
+
+    val IND_SPECTYPE_SELF = Syn.into Syn.IND_SPECTYPE_SELF
+
+    structure ConstrDict = StringListDict
+    structure SpecCtx = Var.Ctx
+    structure DimSet = ListSet (structure Elem = Sym.Ord)
+    type constrs = abt ConstrDict.dict
+    type specctx = abt SpecCtx.dict
+    type dimset = DimSet.set
+
+    val trace = ["InductiveSpec"]
+
+    (* TODO tail recursion *)
+
+    fun EqSpecType H ((ty0, ty1), level) =
+      case (Syn.out ty0, Syn.out ty1) of
+         (Syn.IND_SPECTYPE_SELF, Syn.IND_SPECTYPE_SELF) => []
+       | (Syn.IND_SPECTYPE_FUN (a0, x, b0x), Syn.IND_SPECTYPE_FUN (a1, y, b1y)) =>
+           let
+             val z = Sym.new ()
+             val b0z = VarKit.rename (z, x) b0x
+             val b1z = VarKit.rename (z, y) b1y
+             (* favonia: more research needed for other kinds *)
+             val goalA = makeEq trace H ((a0, a1), Syn.intoU (level, K.COE))
```

Comment:
@ecavallo This is to say that, for `A -> X` to be a spec type, `A` needs to be a coe type.